### PR TITLE
[All hosts] (insider program) Update references to the Office Insider program

### PR DESCRIPTION
**Pull request by samantharamon**
The Office Insider program is now referred to as the Microsoft 365 Insider program (https://insider.microsoft365.com). This PR updates references in our documentation.

### CODE REVIEW

File: docs/outlook/pinnable-taskpane.md [hunk 1]
```diff
@@ -15,8 +15,8 @@ However, by default, if a user has an add-in task pane open for a message in the
 > [!NOTE]
 > Although the pinnable task panes feature was introduced in [requirement set 1.5](/javascript/api/requirement-sets/outlook/requirement-set-1.5/outlook-requirement-set-1.5), it's currently only available to Microsoft 365 subscribers using the following:
 >
-> - Outlook 2016 or later on Windows (build 7668.2000 or later for users in the Current or Office Insider Channels, build 7900.xxxx or later for users in Deferred channels)
-> - Outlook 2016 or later on Mac (version 16.13.503 or later)
+> - Outlook 2016 or later on Windows (Build 7668.2000 or later for users in the Current or Microsoft 365 Insider Channels, Build 7900.xxxx or later for users in Deferred channels)
+> - Outlook 2016 or later on Mac (Version 16.13.503 or later)
 > - Modern Outlook on the web
```
Comment (AlexJerabek) [inline docs/outlook/pinnable-taskpane.md:20]: With Office 2013's obsolescence, do we still need this note?

File: docs/includes/use-legacy-edge-or-ie.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 
 > [!NOTE]
-> The tool that is used to force the change in browser is supported only in the Beta subscription channel of Microsoft 365. Join the [Office Insider program](https://insider.office.com/join/windows) and select the **Beta Channel** option to access Office Beta builds. See also [About Office: What version of Office am I using?](https://support.microsoft.com/office/932788b8-a3ce-44bf-bb09-e334518b8b19).
+> The tool that is used to force the change in browser is supported only in the Beta subscription channel of Microsoft 365. Join the [Microsoft 365 Insider program](https://insider.microsoft365.com/join/windows) and select the **Beta Channel** option to access Office Beta builds. See also [About Office: What version of Office am I using?](https://support.microsoft.com/office/932788b8-a3ce-44bf-bb09-e334518b8b19).
```
Comment (AlexJerabek) [inline docs/includes/use-legacy-edge-or-ie.md:4]: They technically capitalize "Windows" in the URL. In this PR, you have "windows" and "Mac". I don't have a preference for which, but the URL stylings should be consistent.

**Review by AlexJerabek — APPROVED** [review]
Approving with a couple considerations.

File: docs/outlook/pinnable-taskpane.md (hunk 1 above)
Comment (samantharamon) [reply docs/outlook/pinnable-taskpane.md:20]: Thanks for the feedback. I'll follow up on this and make changes in a separate PR, if needed.

File: docs/includes/use-legacy-edge-or-ie.md (hunk 2 above)
Comment (samantharamon) [reply docs/includes/use-legacy-edge-or-ie.md]: Thanks for catching this. Fixed.

**Pull request merged**